tests/fixtures/MakeAuthenticatorLoginFormUserEntityLogout/tests/SecurityControllerTest.php

```diff
@@ -60,10 +62,23 @@ public function testCommand()
         $client->submit($form);
 
         $this->assertStringContainsString('TODO: provide a valid redirect', $client->getResponse()->getContent());
-        $this->assertNotNull($token = $client->getContainer()->get('security.token_storage')->getToken());
-        $this->assertInstanceOf(User::class, $token->getUser());
+        $this->assertInstanceOf(User::class, $this->getToken()->getUser());
 
         $client->request('GET', '/logout');
-        $this->assertNull($client->getContainer()->get('security.token_storage')->getToken());
+        $this->assertNull($this->getToken());
+    }
+
+    /**
+     * Handle Session deprecations in Symfony 5.3+
+     */
+    private function getToken(): ?TokenInterface
+    {
+        $tokenStorage = static::$container->get('security.token_storage');
+
+        if (class_exists(SessionNotFoundException::class)) {
```

Comment:
> And, would it be possible to use a method_exists() instead of the Symfony version?

Yes, kinda. The `disableUsageTracking()` method has always existed in the [UsageTrackingTokenStorage::class](https://github.com/symfony/symfony/blob/4.4/src/Symfony/Component/Security/Core/Authentication/Token/Storage/UsageTrackingTokenStorage.php) since it was introduced in Symfony 4.4

For context, PR 38616 [changed](https://github.com/symfony/symfony/pull/38616/files#diff-2aeaa6c8af14b7abd53c7398c9c7e1bf105541e8470e2f9adc74c2019f81d8b1) how the token is retrieved when calling `UsageTrackingTokenStorage->getToken()` if usage tracking is enabled (which the kernel enables). 

Although I'm not sure we _need_ to, but to support Symfony `4.0` - `4.3` we can't rely on `disableUsageTracking()` to exist. And for Symfony `4.4` - `5.2` I don't think we _should_ disable usage tracking as thats not realistic from the users perspective although we could. E.g. `class_exists(UsageTrackingTokenStorage)`

The `SessionNotFoundException::class` was introduced in `38616` (Symfony `5.3`) and was the only class introduced that didn't look like it would be removed anytime soon.

Reply:
let's keep the kernel check then - thanks for the explantion